src/services/penaltyDocuments.js

```diff
@@ -88,33 +102,150 @@ export default class PenaltyDocument {
 				ID: paymentInfo.id,
 			},
 		};
-		const dbGet = this.db.get(getParams).promise();
-
-		dbGet.then((data) => {
-			data.Item.Value.paymentStatus = paymentInfo.paymentStatus;
-			data.Item.Hash = hashToken(paymentInfo.id, data.Item.Value, data.Item.Enabled);
-			data.Item.Offset = getUnixTime();
-			const putParams = {
+		const docGet = this.db.get(getParams).promise();
+
+		docGet.then((docContainer) => {
+			const doc = docContainer.Item;
+			const newStatus = paymentInfo.paymentStatus;
+			doc.Value.paymentStatus = newStatus;
+			doc.Hash = hashToken(paymentInfo.id, doc.Value, doc.Enabled);
+			doc.Offset = getUnixTime();
+			const docPutParams = {
 				TableName: this.penaltyDocTableName,
-				Item: data.Item,
+				Item: doc,
 				ConditionExpression: 'attribute_exists(#ID)',
 				ExpressionAttributeNames: {
 					'#ID': 'ID',
 				},
 			};
 
-			const dbPut = this.db.put(putParams).promise();
-			dbPut.then(() => {
-				callback(null, createResponse({ statusCode: 200, body: data.Item }));
+			const docPutPromise = this.db.put(docPutParams).promise();
+			let penGrpUpdatePromise;
+			if ((doc.Value.inPenaltyGroup) && (doc.penaltyGroupId && newStatus === 'UNPAID')) {
```

Comment:
Is this case used? I thought that penalty group cancellations were handled by `updateMultipleUponPaymentDelete`?

Reply:
Yep you're right - I'll remove this.

Reply:
Reverted to original version. Now only change to `updateDocumentUponPaymentDelete` is a fix to the error response.